socorro/schemas/socorro-data-1-0-0.schema.yaml

```diff
@@ -0,0 +1,154 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+---
+$schema: http://json-schema.org/draft-07/schema#
+$id: moz://mozilla.org/schemas/socorro/socorro-data/1-0-0
+title: Socorro Data
+description: >
+  Schema for defining Socorro-related data schemas.
+
+definitions:
+  simple_types:
+    enum:
+      - "array"
+      - "boolean"
+      - "integer"
+      - "null"
+      - "number"
+      - "object"
+      - "string"
+
+  medium_id:
+    allOf:
+      - maxLength: 40
```

Comment:
glean-parser's metrics schema enforces snake_case as well, but we can't do that because we have some fields we need to support that aren't snake_case. Similarly, we have longer keys to support, so we use 40 instead of 30.